### PR TITLE
Add logrotation for nginx

### PR DESCRIPTION
**Pull request by robbkidd**
nginx is configured to log to the standard access.log and error.log.
Convincing nginx to log to console in such a way to make runit's svlogd
able to manage it got more complicated than just adding logrotate to
noodle bowl. Precent for logrotate and nginx set in the Chef server, too.
- adds logrotate to the omninbus project to keep it separate from whatever
  logrotate might come on the host
- adds a log_management recipe to configure logrotate and schedule the
  custom supermarket logrotate to run via cron.hourly
- adds a logrotate rule to the nginx cookbook to lay down management
  of nginx's logs and the sending of USR1 to the nginx service after log
  rotation
- tests!

### CODE REVIEW

**Comment by robbkidd** [comment]
PR of work related to chef/supermarket#1141

**Comment by robbkidd** [comment]
New information on nginx and pid files has me rethinking the postrotate call to runit's `sv`. Investigating.

**Comment by robbkidd** [comment]
Rotated by logrotate and nginx logging to new file post-rotate:

```
vagrant@default-ubuntu-1204:~$ sudo ls -ltr /var/log/supermarket/nginx/
total 589840
-rw-r--r-- 1 root        root        38 Oct 27 17:19 config
-rw------- 1 root        root         0 Oct 27 17:19 lock
-rw-r--r-- 1 root        root         0 Oct 27 17:19 current
-rw-r--r-- 1 supermarket root         0 Oct 27 17:19 error.log
-rw-r--r-- 1 supermarket root 603981403 Oct 27 17:48 access.log.1
-rw-r--r-- 1 supermarket root      1619 Oct 27 18:17 access.log
```

@nellshamrell Here 'tis!

@smith @cwebberOps Could I ask for your eyeballs on this as I progress on my omnibus vision quest?

I did what I could to serverspec the logrotation rule short of figuring out how

* to fill the access.log past the rotation size threshold
* then manipulating time to confirm the log gets rotated
* a web request to confirm the nginx process is logging to the new log file

That seems like the log rotation equivalent to "a process named X is listening on port Y," but might be overkill to implement.

Also, this log rotation problem exists in any omnibus installer that includes nginx because nginx logging and runit/svlogd don't get along. Should something be added to the nginx omnibus software component that includes logrotate?

**Comment by nellshamrell** [comment]
This looks reasonable to me.

File: cookbooks/omnibus-supermarket/recipes/log_management.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+# and each component service logged to STDOUT for svlogd to handle. Some
+# components do not handle logging to STDOUT well, so logrotate is added
+# to manage the logs generated by those services.
+
+include_recipe 'omnibus-supermarket::config'
+
+directory "#{node['supermarket']['var_directory']}/etc/logrotate.d" do
+  owner 'root'
+  group 'root'
+  mode '0644'
```
Comment (robbkidd) [inline cookbooks/omnibus-supermarket/recipes/log_management.rb]: This is a directory. It's mode should probably be `0755`. :facepalm:

File: cookbooks/omnibus-supermarket/templates/default/logrotate.cron.erb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#!/bin/sh
+
+PATH=<%= @install_directory %>/embedded/sbin:$PATH
+
+command -v logrotate >/dev/null 2>&1 || exit 0
+nice -n 19 ionice -c3 logrotate <%= @var_directory %>/etc/logrotate.conf
```
Comment (robbkidd) [inline cookbooks/omnibus-supermarket/templates/default/logrotate.cron.erb]: Soooo ... this depends on the system's logrotate being installed. Which was apparently the case in test. Need to add the logrotate dependency to the omnibus project and embed the command.

**Comment by robbkidd** [comment]
I believe this is ready. :shipit: ?

**Pull request merged**